docs/dev/setup/jenkins-localvc.rst

```diff
@@ -0,0 +1,797 @@
+
+   Jenkins is then reachable under ``http://localhost:8082/`` and you can login using the credentials specified
+   in ``jenkins-casc-config-localvc.yml`` (defaults to ``artemis_admin`` as both username and password).
+
+2. The `application-local.yml` must be adapted with the values configured in ``jenkins-casc-config-localvc.yml``:
+
+.. code:: yaml
+
+    artemis:
+        user-management:
+            use-external: false
+            internal-admin:
+                username: artemis_admin
+                password: artemis_admin
+        version-control:
+            url: http://172.17.0.1:8080 # `http://host.docker.internal:8080` for Windows
+            user: root
+            password: dummy # have to be set, but does not matter for LocalVC
+            build-agent-git-username: jenkins
+            build-agent-git-password: artemis_admin # choose some strong password and username (gives read access to all repositories)
+        continuous-integration:
+            user: artemis_admin
+            password: artemis_admin
+            url: http://localhost:8082
+            vcs-credentials: artemis_localvc_credentials
+            artemis-authentication-token-key: artemis_notification_plugin_token
+            artemis-authentication-token-value: artemis_admin
+
+5. Open the ``src/main/resources/config/application-jenkins.yml`` and change the following:
+   Again, if you are using a development setup, the template in the beginning of this page already contains the
+   correct values.
+
+.. code:: yaml
+
+    jenkins:
+        internal-urls:
+            ci-url: http://jenkins:8080
+            vcs-url: http://localhost:8080
+
+6. You're done. You can now run Artemis with the LocalVC/Jenkins environment.
+
+Manual Jenkins Server Setup
+"""""""""""""""""""""""""""
+
+1. Pull the latest Jenkins LTS Docker image
+
+   Run the following command to get the latest jenkins LTS docker image.
+
+   .. code:: bash
+
+       docker pull jenkins/jenkins:lts
+
+2. Create a custom docker image
+
+   In order to install and use Maven with Java in the Jenkins container,
+   you have to first install maven, then download Java and finally
+   configure Maven to use Java instead of the default version.
+   You also need to install Swift and SwiftLint if you want to be able to
+   create Swift programming exercises.
+
+   To perform all these steps automatically, you can prepare a Docker
+   image:
+
+   Create a Dockerfile with the content found `here <docker/jenkins/Dockerfile>`.
+   Copy it in a file named ``Dockerfile``, e.g. in
+   the folder ``/opt/jenkins/`` using ``vim Dockerfile``.
+
+   Now run the command ``docker build --no-cache -t jenkins-artemis .``
+
+   This might take a while because Docker will download Java, but this
+   is only required once.
+
+3. **If you run your own NGINX or if you install Jenkins on a local development computer, then skip the next steps (4-7)**
+
+4. Create a file increasing the maximum file size for the nginx proxy.
```

Comment:
```suggestion
Nginx proxy setup
#################

If you run your own NGINX or if you install Jenkins on a local development computer, you can skip this section.

#. Create a file increasing the maximum file size for the Nginx proxy.
```
Since the two steps above can be removed, this part only contains Nginx setup instructions now and could receive its own sub-heading.